Making the event component

EmperorYP7 opened the pull request:
fixes #112 ~fixes#116~ fixes #115
<Do not delete this template>
Description: <provide a description> Made a tabular structure for displaying all the events.

#### Checklist
- [x] I've followed the Contributing guidelines provided in the repository.
- [x] I've made the changes which were demanded in the linked issue.
- [x] I've tested my code on a Chromium based browser.
- [x] I've tested my code on Mozilla Firefox.
- [x] My code gave a clean console on debugging. (no warnings/errors)

#### Testimonial

https://deploy-preview-191--houseofgeeks.netlify.app/event

![image](https://user-images.githubusercontent.com/62606998/97556224-2e5af580-19ff-11eb-8db3-04a171ed8c1b.png)

Commit:
adding basic structure to event

Commit:
added event table

Commit:
Merge branches 'front-end' and 'front-end' of https://github.com/houseofgeeks/hg into front-end

Commit:
added event table

Commit:
Merge branch 'front-end' of https://github.com/houseofgeeks/hg into front-end

Commit:
improving the design

Commit:
adding responsiveness :iphone:

tusharjain0022 reviewed (commented):
LGTM ! Nice work @EmperorYP7 🤩 
Just a small change that in the date | time column as time is being stored acc. to 12-hr time convention then it is should be printed with AM or PM.

ankiiitraj commented:
Add a column for description and use extrnal_link_icon after the title! 
rest LGTM
something like 

the package used for icons https://react-icons.github.io/react-icons/

![Screenshot from 2020-10-29 01-44-20](https://user-images.githubusercontent.com/48787278/97491558-565b4200-1988-11eb-9fea-31aec36fd1b2.png)

anktrj commented on client/src/components/Event/Tables/UpcomingEvents.js:
```diff
@@ -0,0 +1,63 @@
+import React, { useEffect, useState } from 'react';
+import "bootstrap/dist/css/bootstrap.min.css";
+import './UpcomingEvents.css';
+
```
add 
` import { BiLinkExternal } from 'react-icons/bi'; `

anktrj commented on client/src/components/Event/Tables/UpcomingEvents.js:
```diff
@@ -0,0 +1,63 @@
+                                                href={`${event.eventLink}`}
+                                                target="_blank"
+                                            >
+                                                {`${event.title}`}
```
add 
`<BiLinkExternal/>`
here for the icon

ankiiitraj reviewed (commented):
Also, try adding a message to show while the evnts are being fetched!

Commit:
12 hr format & icon

ankiiitraj commented:
The columns are not aligned and look odd! What do you say @hariom1625 @tusharjain0022 
![Screenshot from 2020-10-29 11-57-15](https://user-images.githubusercontent.com/48787278/97533262-e7a8d380-19dd-11eb-80c2-27c565e8fe84.png)

hariom1625 commented:
> The columns are not aligned and look odd! What do you say @hariom1625 @tusharjain0022

I too noticed it does look a bit odd and also the prior 24 hrs date and time format was alright because the present commit is quite confusing maybe it isn't rendering properly.
And if we will be using 24hrs format, printing AM and PM wouldn't be necessary.

EmperorYP7 commented:
> The columns are not aligned and look odd! What do you say @hariom1625 @tusharjain0022

I am working my way around it. If anyone figures it out, you're welcome to push in some code here. :wink:


> I too noticed it does look a bit odd and also the prior 24 hrs date and time format was alright because the present commit is quite confusing maybe it isn't rendering properly.
> And if we will be using 24hrs format, printing AM and PM wouldn't be necessary.

I've fixed it, will be pushing it soon. It was something wrong with the way I implemented the function to convert 24hr format to 12hr format which didn't work out as expected in firefox.

Commit:
minor fixes

Commit:
Merge branches 'front-end' and 'front-end' of https://github.com/houseofgeeks/hg into front-end

tusharjain0022 reviewed (approved):
LGTM! 🤩

ankiiitraj commented:
@EmperorYP7 is it ready to merged?

EmperorYP7 commented:
![image](https://user-images.githubusercontent.com/62606998/97598359-e2bf4080-1a2c-11eb-97e7-b5f183192083.png)

I am noticing some wierd margin at the top. Couldn't figure out where its coming from. But this PR can be merged. We can make an issue to solve this.

tusharjain0022 commented on client/src/components/Event/Event.css:
```diff
@@ -1,12 +1,47 @@
     color: #EE2A7B;
     font-size: 8vh;
     opacity: 1;
     margin: 5vh 0;
```
here the top margin of `.event-title` is causing that unnecessary margin on the top. it should be `margin-bottom: 5vh ;`

tusharjain0022 replied:
I would suggest that this issue can be added to #195 with other minor updates or we can open a separate issue for it with issue of the alignment of columns of the table. and this PR is ready to be merged as here's already a lot of work done.

EmperorYP7 replied:
That seems to be working for now. Thanks for pointing out @tusharjain0022! :smile:

hariom1625 reviewed (approved):
LGTM !!
Great work @EmperorYP7  :-)

Commit:
fixed margin issue
